src/pdhg.jl

```diff
@@ -0,0 +1,142 @@
+# based on https://github.com/klamike/SimplePDHG.jl
+
+function pdhg_handler(lines::Vector{Vector{Float64}}, objective::Vector{Float64})
+    m = length(lines)
+    A = zeros(m, 2)
+    b = zeros(m)
+    for i in 1:m
+        A[i, 1] = lines[i][1]
+        A[i, 2] = lines[i][2]
+        b[i] = lines[i][3]
+    end
+    return solve_pdhg(A, b, objective)
+end
+
+
+struct PDHGProblem{T<:Real,I<:Integer}
+    c::AbstractVector{T}
+    A::AbstractMatrix{T}
+    b::AbstractVector{T}
+    m::I # number of rows of A
+    n::I # number of columns of A
+end
+
+mutable struct PDHGState{T<:AbstractVecOrMat{<:Real},I<:Integer} # contains information regarding one iterattion sequence
+    x::T # iterate x_n
+    y::T # iterate y_n
+    z::T # iterate z_n
+    η::Real # step size
+    τ::Real # step size
+    k::I # iteration counter
+end
+
+function PDHGState(problem::PDHGProblem{T,I}) where {T<:Real,I<:Integer}
+    n = problem.n
+    m = problem.m
+    σmaxA = LinearAlgebra.norm(problem.A, 2)
+    η_preli = (1 / (σmaxA)) - 1e-6
+    τ_preli = (1 / (σmaxA)) - 1e-6
+    @assert η_preli > 0 && τ_preli > 0 "Got negative initial step sizes"
+    x_0 = zeros(T, n)
+    y_0 = zeros(T, m)
+    return PDHGState(x_0, y_0, problem.c, η_preli, τ_preli, 1)
+end
+
+function project_nonnegative!(x::AbstractVector{T}) where {T<:Real}
+    for i in eachindex(x)
+        x[i] = max(zero(T), x[i])
+    end
+end
+
+function project_nonnegative(x::AbstractVector{T}) where {T<:Real}
+    y = zeros(length(x))
```

Comment:
`zeros` will return a `Float64`-valued array by default
```suggestion
    y = zeros(T, length(x))
```